[yucongkoo] iP

yucongkoo opened the pull request:
## Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke has the following benefits:
* text-based
* easy-to-use
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is:
1. dowload the .jar file from [here](https://github.com/yucongkoo/ip/releases/tag/A-Gradle)
1. run the jar file using ```java -jar duke.jar```
1. add your tasks
1. let it manage your tasks for you 😄 

And it is **FREE!**

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon) 
- [ ] Reminders (coming soon)

***

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yucongkoo committed:
Implementation of Level-0 features

yucongkoo committed:
Implementation of Level-1 features

yucongkoo committed:
Implementation of Level-2 features

yucongkoo committed:
Implementation of Level-3 features

yucongkoo committed:
Implementation of Level-4 features

yucongkoo committed:
Implementation of Level-6 features

yucongkoo committed:
Implementation of A-TextUiTesting

yucongkoo committed:
Functionality of getting target file

yucongkoo committed:
Merge branch 'add-gradle-support' of https://github.com/yucongkoo/ip

yucongkoo committed:
Merge branch 'master' into branh-Level-7

yucongkoo committed:
Add Level-7 features

yucongkoo committed:
Merge branch 'branh-Level-7'

yucongkoo committed:
Add DiskManager

yucongkoo committed:
Add level-8 features

yucongkoo committed:
Merge branch 'branch-Level-8'

yucongkoo committed:
Add A-MoreOOP feature

yucongkoo committed:
Merge branch 'branch-A-MoreOOP'

yucongkoo committed:
Add A-Gradle features

Added gradle support to run the application.

yucongkoo committed:
Update the runtest.sh script to use gradle commands

yucongkoo committed:
Add error handling when saving to local disk

Previously did not handle error when saving to local disk

Unexpected values will be printed

Lets add error handling so the application can handle such errors

yucongkoo committed:
Add A-JavaDoc features

Added javadoc to all public method and classes.

yucongkoo committed:
Add A-CodingStandard features

Tweaked code to follow coding standards

yucongkoo committed:
Add Level-9 features.

Implemented the find command to match certain patterns in task list.

yucongkoo committed:
Merge branch 'branch-A-JavaDoc'

yucongkoo committed:
Merge branch 'branch-A-CodingStandard'

yucongkoo committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

yucongkoo committed:
Remove unicode characters

Removed unicode characters since displaying them is inconsistent

yucongkoo committed:
Add Test file

Added dummy test file, unable to import junit

yucongkoo committed:
Add JUnit test for loadFromDisk method in DiskManager

Added Junit test to test the loadFromDisk method extensively.

yucongkoo committed:
Add JUnit test for parseCommand method in Parser

Added Junit test to test the parseCommand method extensively.

yucongkoo committed:
Merge branch 'branch-A-JUnit'

yucongkoo committed:
Implement framework for GUI

Followed the tutorial to implement the skeleton for GUI

yucongkoo committed:
Add Level-10 features

Added features for level 10 incremenet for basic functioning gui

yucongkoo committed:
Remove redundant data

Data is not supposed to be in the branch

yucongkoo committed:
Fix bug

Fixed the bug where test resource has incorrect format

yucongkoo committed:
Merge branch 'branch-Level-10'

yucongkoo committed:
Add trivial stuff

Added some trivial stuff such as adding title and setting main class for gradle

yucongkoo committed:
Add styling to GUI

Added some styling to the GUI

yucongkoo committed:
Add assertions to DiskManager

There are several methods in DiskManager that made assumptions to
certain values.

Using assertions to ensure such values fulfil its requirements guards the
program from unexpected errors.

Let's add assertions to these values.

yucongkoo committed:
Add assertions to remaining files

Added assertions to remaning files to guard our program from
unexpected errors.

yucongkoo committed:
Refactor code to improve code quality

There are certain parts of the code that could be a little hard
to understand the logic flow.

Refactor code to improve the code quality, making the code
more readable and understandable.

Let's refactor code segments that could be improved.

yucongkoo committed:
Merge pull request #1 from yucongkoo/branch-A-Assertions

Add assertions

yucongkoo committed:
Merge branch 'master' of https://github.com/yucongkoo/ip

yucongkoo committed:
Merge branch 'master' into branch-A-CodeQuality

yucongkoo committed:
Merge pull request #2 from yucongkoo/branch-A-CodeQuality

Refactor code to improve code quality

yucongkoo committed:
Add extension C-BetterSearch

yucongkoo committed:
Merge branch 'branch-BCD-Extension'

joeng03 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,207 @@
+        }
+
+        if (command.equals(FLAG_MARK)) {
+            Scanner tempSc = new Scanner(value);
+
+            if (!tempSc.hasNextInt()) {
+                tempSc.close();
+                throw new DukeException("Oops!!! Invalid argument of a mark command");
+            }
+
+            int index = tempSc.nextInt();
+            if (tempSc.hasNext()) {
+                // means invalid formatting for done command
+                tempSc.close();
+                throw new DukeException("Oops!!! Invalid argument of a mark command");
+            }
+            tempSc.close();
```
May want to abstract out the code handling inputs and their exceptions (Don't repeat yourself)

joeng03 commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,136 @@
+        StringBuilder sb = new StringBuilder();
+        sb.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); ++i) {
+            String item = (i + 1) + "." + tasks.get(i).toString() + "\n";
```
For brevity, ` String item = (i + 1) + "." + tasks.get(i) + "\n"; ` is sufficient as Java is smart ;)) 
Similarly for other instances involving concatenation of `Object`s

joeng03 commented on src/main/java/duke/TaskManager.java:
```diff
@@ -0,0 +1,136 @@
+        }
+
+        Task task = this.tasks.remove(i);
+        return "Noted! I've removed this task:\n  "
+                + task.toString() + "\n"
+                + "Now you have " + this.tasks.size() + " tasks in the list.\n";
```
Might want to do a separation of concerns by defining these print methods in a separate Ui class to avoid magic literals.

joeng03 reviewed (commented):
Great job overall! Nice usage of libraries, solid OOP, and high test coverage. Coding standards were also well adhered to.

yucongkoo committed:
Refactor long code into multiple methods

Refactored code to follow coding qualities

yucongkoo committed:
Fix regression caused by assertions

Fixed the regression caused by assertions, causing the
loadFromDisk() method failing

yucongkoo committed:
Fix Parser Bug

yucongkoo committed:
Modify unit test test case

yucongkoo committed:
Add user guide

Added user guide

yucongkoo committed:
Add product screenshot

yucongkoo committed:
Resolve code quality issue

Resolved the code quality issues proposed by nus-se-script

yucongkoo committed:
Modify docs/README.md

Modified the user guide format
